cmd/server/app/server.go

```diff
@@ -0,0 +1,372 @@
+		client.RegisterProxyServiceServer(grpcServer, s)
+		lis, err := net.Listen("tcp", addr)
+		if err != nil {
+			return nil, fmt.Errorf("failed to listen on %s: %v", addr, err)
+		}
+		go grpcServer.Serve(lis)
+		stop = grpcServer.GracefulStop
+	} else {
+		// http-connect
+		server := &http.Server{
+			Addr:      addr,
+			TLSConfig: tlsConfig,
+			Handler: &server.Tunnel{
+				Server: s,
+			},
+			TLSNextProto: make(map[string]func(*http.Server, *tls.Conn, http.Handler)),
+		}
+		stop = func() {
+			err := server.Shutdown(ctx)
+			if err != nil {
+				klog.ErrorS(err, "failed to shutdown server")
+			}
+		}
+		go func() {
+			err := server.ListenAndServeTLS("", "") // empty files defaults to tlsConfig
+			if err != nil {
+				klog.ErrorS(err, "failed to listen on master port")
+			}
+		}()
+	}
+
+	return stop, nil
+}
+
+func (p *Proxy) runAgentServer(o *options.ProxyRunOptions, server *server.ProxyServer) error {
+	var tlsConfig *tls.Config
+	var err error
+	if tlsConfig, err = p.getTLSConfig(o.ClusterCaCert, o.ClusterCert, o.ClusterKey); err != nil {
+		return err
+	}
+
+	addr := fmt.Sprintf(":%d", o.AgentPort)
+	serverOptions := []grpc.ServerOption{
+		grpc.Creds(credentials.NewTLS(tlsConfig)),
+		grpc.KeepaliveParams(keepalive.ServerParameters{Time: o.KeepaliveTime}),
+	}
+	grpcServer := grpc.NewServer(serverOptions...)
+	agent.RegisterAgentServiceServer(grpcServer, server)
+	lis, err := net.Listen("tcp", addr)
+	if err != nil {
+		return fmt.Errorf("failed to listen on %s: %v", addr, err)
+	}
+	go grpcServer.Serve(lis)
+
+	return nil
+}
+
+func (p *Proxy) runAdminServer(o *options.ProxyRunOptions, server *server.ProxyServer) error {
+	muxHandler := http.NewServeMux()
+	muxHandler.Handle("/metrics", promhttp.Handler())
+	if o.EnableProfiling {
+		muxHandler.HandleFunc("/debug/pprof", util.RedirectTo("/debug/pprof/"))
+		muxHandler.HandleFunc("/debug/pprof/", pprof.Index)
+		if o.EnableContentionProfiling {
+			runtime.SetBlockProfileRate(1)
+		}
+	}
+	adminServer := &http.Server{
+		Addr:           fmt.Sprintf("127.0.0.1:%d", o.AdminPort),
+		Handler:        muxHandler,
+		MaxHeaderBytes: 1 << 20,
+	}
+
+	go func() {
+		err := adminServer.ListenAndServe()
+		if err != nil {
+			klog.ErrorS(err, "admin server could not listen")
+		}
+		klog.V(1).Infoln("Admin server stopped listening")
+	}()
+
+	return nil
+}
+
+func (p *Proxy) runHealthServer(o *options.ProxyRunOptions, server *server.ProxyServer) error {
+	livenessHandler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		fmt.Fprintf(w, "ok")
+	})
+	readinessHandler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		ready, msg := server.Readiness.Ready()
+		if ready {
+			w.WriteHeader(200)
+			fmt.Fprintf(w, "ok")
+			return
+		}
+		w.WriteHeader(500)
+		fmt.Fprintf(w, msg)
+	})
+
+	muxHandler := http.NewServeMux()
+	muxHandler.HandleFunc("/healthz", livenessHandler)
+	muxHandler.HandleFunc("/ready", readinessHandler)
```

Comment:
I realize this was a pre-existing bug but this should be ```/readyz```.

Reply:
I found that it is /ready in master branch. Maybe we can fix it in other PRs. This PR only refine main package.